src/output/tree.rs

```diff
@@ -142,6 +139,17 @@ impl<'module> MonoidalPrinter<'module> {
                     }
                 }
             }
+            Value::Branch(n, value) => match format {
+                Some(Format::Union(branches)) => {
+                    let format = &branches[*n];
+                    self.is_atomic_value(value, Some(format))
+                }
+                Some(Format::UnionVariant(branches)) | Some(Format::UnionNondet(branches)) => {
+                    let format = &branches[*n].1;
+                    self.is_atomic_value(value, Some(format))
+                }
+                _ => self.is_atomic_value(value, None),
```

Comment:
Do we care about the difference between seeing `None` vs. `Some(_x)` here, where `_x` is implicitly a non-union-oriented format? In other words, is an unexpected format in this position deemed a panic-worthy error, or is it sensible to gloss over those cases?